### DISCUSSION TITLE
Multi-Source Control System Support

### DISCUSSION THREAD

**Discussion by cjdutoit** [The-Standard-Organization/GitFyle.Core.Api#2] [Ideas]
@JayserMendez: The product must support integration with various source control systems like GitHub, Azure DevOps, SVN, Mercurial etc.

**Comment by makhammadsoliyev** [comment]
> @JayserMendez: The product must support integration with various source control systems like GitHub, Azure DevOps, SVN, Mercurial etc.

![image](https://github.com/The-Standard-Organization/GitFyle.Core.Api/assets/149594973/47c27416-0b72-4b75-98e5-ddb68f35d28f)
If there must be a unique constraint on (RepositoryId, ExternalId), what if there is a new contribution with the same RepositoryId and ExternalId, but different ContributorId and/or ContributionTypeId. Can you clarify it?
